### PR TITLE
TSDB: shrink _id inverted index

### PR DESCRIPTION
**Pull request by nik9000**
This shrinks the inverted index for the `_id` field in tsdb by ~37% by
encoding the `@timestamp` portion of the `_id` big endian. Previously
the least significant bits were early in the term which frustred the
prefix-encoded inverted index.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/common/util/ByteUtils.java [hunk 1]
```diff
@@ -41,6 +43,16 @@ public static long readLongLE(byte[] arr, int offset) {
         return (long) LITTLE_ENDIAN_LONG.get(arr, offset);
     }
 
+    /** Write a long in little-endian format. */
```
Comment (tobiasstadler) [inline server/src/main/java/org/elasticsearch/common/util/ByteUtils.java:46]: Shouldn't little-endian be go-endian?

Comment (nik9000) [reply server/src/main/java/org/elasticsearch/common/util/ByteUtils.java:46]: :+1:

**Comment by nik9000** [comment]
@elasticmachine, update branch

**Comment by nik9000** [comment]
@elasticmachine update branch

**Comment by nik9000** [comment]
> @elasticmachine update branch

OK robot, I'll do it myself.

**Review by romseygeek — APPROVED** [review]
LGTM!

**Comment by jpountz** [comment]
Great!

**Pull request merged**